modules/libcom/RTEMS/posix/rtems_init.c

```diff
@@ -0,0 +1,1159 @@
+{
+    rtems_shutdown_executive(0);
+}
+
+#ifndef RTEMS_LEGACY_STACK
+static char* getPrimaryNetworkInterface(void)
+{
+    // lookup available network interfaces
+    char ifnamebuf[IF_NAMESIZE];
+    char *ifname;
+    // get primary network interface
+    ifname = if_indextoname(1, &ifnamebuf[0]);
+    if (ifname == NULL) return (NULL);
+    printf("\n***** Primary Network interface : %s *****\n", ifname);
+    return (ifname);
+}
+
+static void
+dhcpcd_hook_handler(rtems_dhcpcd_hook *hook, char *const *env)
+{
+    int bound = 0;
+    char iName[16];
+    char *name;
+    char *value;
+
+	(void)hook;
+
+    char ifnamebuf[IF_NAMESIZE];
+    sprintf(ifnamebuf, "%s", getPrimaryNetworkInterface());
+
+    while (*env != NULL) {
+        name = strtok(*env,"=");
+        value = strtok(NULL,"=");
+        printf("all out ---> %s = %s\n", name, value);
+        if (!strncmp(name, "interface", 9) && !strcmp(value,  ifnamebuf))
+			 strncpy(iName, value, 16);
+		if (!strncmp(name, "reason", 6) && !strncmp(value, "BOUND", 5)){
+            printf ("Interface %s bounded\n", iName);
+            bound = 1;
+		}
+		if (bound) {
+            // as there is no ntp-support in rtems-libbsd, we call our own client
+            if(!strncmp(name, "new_ntp_servers", 15))
+                strcpy(rtemsInit_NTP_server_ip,value);
+            if(!strncmp(name, "new_host_name", 13))
+                sethostname (value, strlen (value));
+            if(!strncmp(name, "new_tftp_server_name", 20)){
+                //printf(" new_tftp_server_name : %s\n", value);
+                strncpy(rtems_bsdnet_bootp_server_name,value, sizeof(bootp_server_name_init));
+                printf(" rtems_bsdnet_bootp_server_name : %s\n", rtems_bsdnet_bootp_server_name);
+            }
+            if(!strncmp(name, "new_bootfile_name", 20)){
+                //printf(" new_bootfile_name : %s\n", value);
+                strncpy(rtems_bsdnet_bootp_boot_file_name,value, sizeof(bootp_boot_file_name_init));
+                printf(" rtems_bsdnet_bootp_boot_file_name : %s\n", rtems_bsdnet_bootp_boot_file_name);
+            }
+            if(!strncmp(name, "new_rtems_cmdline", 20)){
+                //printf(" new_rtems_cmdline : %s\n", value);
+                strncpy(rtems_bsdnet_bootp_cmdline,value, sizeof(bootp_cmdline_init));
+                printf(" rtems_bsdnet_bootp_cmdline : %s\n", rtems_bsdnet_bootp_cmdline);
+            }
+            // printf("---> %s = %s\n", name, value);
+        }
+		++env;
+	}
+    if (bound)
+        epicsEventSignal(dhcpDone);
+    }
+
+static rtems_dhcpcd_hook dhcpcd_hook = {
+	.name = "ioc boot",
+	.handler = dhcpcd_hook_handler
+};
+
+static void
+default_network_on_exit(int exit_code, void *arg)
+{
+        rtems_printer printer;
+
+        (void)arg;
+
+        rtems_print_printer_printf(&printer);
+        rtems_stack_checker_report_usage_with_plugin(&printer);
+
+        if (exit_code == 0) {
+                puts("*** END OF TEST ***");
```

Comment:
Not really a functional issue, but someone is certain to ask if this appears during a production IOC reboot.

Reply:
I removed it... presumably its not desired?